### PR TITLE
freebsd.libc: break into many small derivations

### PR DESCRIPTION
**Pull request by rhelmot**
The freebsd libc derivation is this monster which compiles many many libraries. This breaks it up into many small derivations and symlinkjoins them together.

Most of this diff is just the new files and the destruction of the big libc file. There are some knock-on effects that some of these libraries were already packaged but only for the bootstrapping (host != freebsd) cases. These were renamed to have a -boot suffix and dependent packages were adjusted.

## Things done

- Built on platform(s)
  - [x] x86_64-linux (freebsd cross)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Ericson2314** [comment]
Very much like splitting it up! But can we do this without `-boot` though? Just adding some conditional stuff to those packages should work fine.

File: pkgs/os-specific/bsd/freebsd/pkgs/libgcc.nix
Comment (Ericson2314) [inline pkgs/os-specific/bsd/freebsd/pkgs/libgcc.nix:1]: Is this significantly diverged from upstream compiler-rt (which can also make a libgcc compat thing now)?

Comment (Ericson2314) [reply pkgs/os-specific/bsd/freebsd/pkgs/libgcc.nix:1]: (But this was already in use before the split up, so consider this just a question, not a PR blocker. Stopping using this is indeed an orthogonal change best left for a future PR.)

**Comment by rhelmot** [comment]
It would be helpful if https://github.com/NixOS/nixpkgs/pull/337351 were merged before we work on non-hacky solutions for -boot.

**Comment by rhelmot** [comment]
Debooted. This entailed adding some more patches to make the makefiles work as expected in both cases (mostly related to installing with chown). I had to merge this with a different patch that touched the same file for a different reason, so the diff looks a little funky.

**Review by Ericson2314 — APPROVED** [review]
Thanks!

**Comment by Ericson2314** [comment]
@ofborg eval

**Pull request merged**